earthfile2llb/earthfile2llb.go

```diff
@@ -198,6 +198,9 @@ type ConvertOpt struct {
 
 	// NoAutoSkip disables auto-skip usages.
 	NoAutoSkip bool
+
+	// RootTarget tracks the topmost target used to invoke the build.
+	RootTarget domain.Target
```

Comment:
I'm trying to understand if this is really needed, since the non-glob resolving doesn't have something equivalent. Also, what happens if there's a chain of more than 2 relative target references? Would the logic yield the correct reference after going through the chain?

See the logic here for the non-glob case: https://github.com/earthly/earthly/blob/main/domain/reference.go#L57

Could we adapt that logic to make it work for the glob case too instead?

Reply:
It's probably not needed. I wasn't sure of the best way to derive the topmost target's path.

What do you mean by "chain" in this context? Can you give me an example?

Reply:
Yeah, like `BUILD ./something+foo` then calling `BUILD ../something-else+bar` then calling `BUILD ./my-dir/*+buz`. Maintaining the correct relative path shouldn't depend on the path of the root target, but rather you need to keep track of it every step of the way. This can get a bit complex when remote references are thrown in the mix. That's why I would reuse the code for JoinReferences, because it's been tried and tested to keep track of this stuff for the non-glob case.

Reply:
Should be good for another look.